### PR TITLE
Updated to use the dataset API 

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Updated to use the dataset API where endpoints have been moved from the import API.

Split the existing API class into two - the ImportAPI and the DatasetAPI.

### How to review

Review code updates.

### Who can review

Anyone

### CODE REVIEW

File: api/dataset_api.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+type InstanceResults struct {
+	Items []Instance `json:"items"`
+}
+
+// Instance comes in results from the Dataset API
+type Instance struct {
+	InstanceID                string `json:"instance_id"`
+	NumberOfObservations      int64  `json:"total_observations"`
+	TotalInsertedObservations int64  `json:"total_inserted_observations,omitempty"`
+	Job                       IDLink `json:"job"`
```
Comment (Matt-Guest) [inline api/dataset_api.go]: The json in the dataset-api has changes to 
```
{
	"links": {
		"job": {
			"id": "123-456",
			"href": "http://localhost:2200/jobs/123-456"
		}
	}
}
```

Comment (CarlHembrough) [reply api/dataset_api.go]: I have updated to use the latest model 👍

File: api/dataset_api.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+// DatasetAPI aggregates a client and URL and other common data for accessing the API
+type DatasetAPI struct {
+	Client    *http.Client
+	URL       string
+	AuthToken string
+}
+
+// NewDatasetAPI creates an DatasetAPI object
+func NewDatasetAPI(client *http.Client, url string, authToken string) *DatasetAPI {
```
Comment (eldeal) [inline api/dataset_api.go]: This constructor isnt needed if its just setting the public vars

File: api/import_api.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+// InstanceLink identifies an (instance or import-job) by id and URL (from Import API)
+type InstanceLink struct {
+	ID   string `json:"id"`
+	Link string `json:"link"`
+}
+
+// NewImportAPI creates an ImportAPI object
+func NewImportAPI(client *http.Client, url, authToken string) *ImportAPI {
```
Comment (eldeal) [inline api/import_api.go]: Constructor not needed?

File: cmd/dp-import-tracker/main.go [hunk 4]
```diff
@@ -62,34 +64,34 @@ func (trackedInstances trackedInstanceList) updateInstanceFromImportAPI(api *api
...
 		tempCopy.jobID = importID
 	}
 	trackedInstances[instanceID] = tempCopy
 	return nil
 }
 
 // getInstanceListFromImportAPI gets a list of current import Instances, to seed our in-memory list
-func (trackedInstances trackedInstanceList) getInstanceListFromImportAPI(api *api.ImportAPI) error {
-	instancesFromAPI, err := api.GetInstances(url.Values{"instance_states": []string{"created"}})
+func (trackedInstances trackedInstanceList) getInstanceListFromImportAPI(datasetAPI *api.DatasetAPI) error {
```
Comment (eldeal) [inline cmd/dp-import-tracker/main.go]: Function name still says import. 

Also probably not necessary to rename the var from api to datasetAPI though, as the function definition provides context as to what type the api is. Not a major issue, just idiomatic to tend towards shorter var names where reasonable.

File: cmd/dp-import-tracker/main.go [hunk 5]
```diff
@@ -117,24 +119,29 @@ func CheckImportJobCompletionState(api *api.ImportAPI, jobID, completedInstanceI
...
 		log.ErrorC("CheckImportJobCompletionState update", err, log.Data{"jobID": jobID, "last completed instanceID": completedInstanceID})
 	}
 	return nil
 }
 
 // updateInstanceWithObservationsInserted updates a specific import instance with the counts of inserted observations
-func updateInstanceWithObservationsInserted(api *api.ImportAPI, instanceID string, observationsInserted int32) error {
-	if err := api.UpdateInstanceWithNewInserts(instanceID, observationsInserted); err != nil {
+func updateInstanceWithObservationsInserted(datasetAPI *api.DatasetAPI, instanceID string, observationsInserted int32) error {
+	if err := datasetAPI.UpdateInstanceWithNewInserts(instanceID, observationsInserted); err != nil {
```
Comment (eldeal) [inline cmd/dp-import-tracker/main.go]: dont need the if, can just `return datasetAPI.Update...` (though at that point this whole function is superfluous)

File: api/dataset_api.go (hunk 2 above)
Comment (CarlHembrough) [reply api/dataset_api.go]: The vars should not have been public. I have made them private.

File: api/import_api.go (hunk 3 above)
Comment (CarlHembrough) [reply api/import_api.go]: The vars should not have been public. I have made them private.

File: cmd/dp-import-tracker/main.go (hunk 4 above)
Comment (CarlHembrough) [reply cmd/dp-import-tracker/main.go]: I have taken the API name out of the function name as it did not seem necessary. I have also renamed datasetAPI to api as suggested.

File: cmd/dp-import-tracker/main.go (hunk 5 above)
Comment (CarlHembrough) [reply cmd/dp-import-tracker/main.go]: Good point. I have inlined the function 👍

**Pull request merged**